### PR TITLE
CBL-689: Rip out old implementation of OFFLINE

### PR DESCRIPTION
**Pull request by bmeike**
This commit removes all of the old platform logic for the OFFLINE state.  It also disables the NetworkReachabilityManager.  CBL-760 tracks re-enabling it

### CODE REVIEW

File: common/main/java/com/couchbase/lite/AbstractReplicator.java [hunk 1]
```diff
@@ -70,12 +66,9 @@
  * be notified of progress.
  */
 @SuppressWarnings({"PMD.GodClass", "PMD.CyclomaticComplexity", "PMD.TooManyMethods", "PMD.TooManyFields"})
-public abstract class AbstractReplicator extends NetworkReachabilityListener {
+public abstract class AbstractReplicator {
```
Comment (pasin) [inline common/main/java/com/couchbase/lite/AbstractReplicator.java]: I think NetworkReachabilityListener is still needed. Basically the platform still needs to listener to the reachability change when the replicator's status of OFFLINE and inform the C4Replicator that the host becomes reachable via c4repl_setHostReachable() API.

File: common/test/java/com/couchbase/lite/ReplicatorOfflineTest.java [hunk 2]
```diff
@@ -105,6 +106,7 @@ public void testChangeListenerEmptyArg() throws Exception {
     }
 
     @Test
+    @Ignore("Platform no longer has control of network reachablity")
```
Comment (pasin) [inline common/test/java/com/couchbase/lite/ReplicatorOfflineTest.java]: Platform is still responsible for controlling network reachability. Platform needs to watch for the reachability when the replicator is OFFLINE and inform the C4Replicator to retry when the host is reachable.

Comment (bmeike) [reply common/test/java/com/couchbase/lite/ReplicatorOfflineTest.java]: As noted in the description of this ticket, CBL-760

I though you all were complaining that my commits were to big.  Here's one divided into two, reasonable steps...

File: common/main/java/com/couchbase/lite/AbstractReplicator.java (hunk 1 above)
Comment (pasin) [reply common/main/java/com/couchbase/lite/AbstractReplicator.java]: I just saw CBL-760.

**Pull request merged**